### PR TITLE
zigbee2mqtt: 1.16.1 -> 1.16.2

### PR DESCRIPTION
**Pull request by zaninime**
###### Motivation for this change

https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.16.2

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

/cc @mweinelt @sweber83

### CODE REVIEW

File: pkgs/servers/zigbee2mqtt/update.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env nix-shell
-#! nix-shell -i bash -p nodePackages.node2nix nodejs-12_x curl jq nix-update
+#! nix-shell -i bash -p nodePackages.node2nix nodejs-12_x curl jq nix-update wget
```
Comment (mweinelt) [inline pkgs/servers/zigbee2mqtt/update.sh:2]: We might as well rewrite this to use curl, which is already in there.

File: pkgs/servers/zigbee2mqtt/default.nix [hunk 2]
```diff
@@ -3,14 +3,15 @@ let
...
 package.override rec {
-  version = "1.16.1";
+  version = "1.16.2";
   reconstructLock = true;
 
   postInstall = ''
     sed -i '1s;^;#!/usr/bin/env node\n;' $out/lib/node_modules/zigbee2mqtt/index.js
-    chmod +x $out/lib/node_modules/zigbee2mqtt/index.js
+    chmod +x $out/lib/node_modules/zigbee2mqtt/cli.js
+    rm -rf $out/bin
```
Comment (mweinelt) [inline pkgs/servers/zigbee2mqtt/default.nix:12]: What's in here that you need to delete?

**Review by mweinelt — COMMENTED** [review]
Good catch regarding the DATADIR. I have a question and a suggestion.

**Comment by mweinelt** [comment]
@GrahamcOfBorg test zigbee2mqtt

File: pkgs/servers/zigbee2mqtt/default.nix (hunk 2 above)
Comment (zaninime) [reply pkgs/servers/zigbee2mqtt/default.nix:12]: The bin created by npm: https://github.com/Koenkk/zigbee2mqtt/blob/1.16.2/package.json#L79

File: pkgs/servers/zigbee2mqtt/update.sh (hunk 1 above)
Comment (zaninime) [reply pkgs/servers/zigbee2mqtt/update.sh]: Switched to `curl`.

File: pkgs/servers/zigbee2mqtt/default.nix [hunk 3]
```diff
@@ -3,14 +3,15 @@ let
...
     mkdir $out/bin
-    ln -s $out/lib/node_modules/zigbee2mqtt/index.js $out/bin/zigbee2mqtt
+    ln -s $out/lib/node_modules/zigbee2mqtt/cli.js $out/bin/zigbee2mqtt
```
Comment (makefu) [inline pkgs/servers/zigbee2mqtt/default.nix:14]: any reason not to use upstream generated binary? that way the chmod, rm,mkdir and ln commands would be come obsolete.
or is upstream bin/zigbee2mqtt symlink not working afterwards?

Comment (mweinelt) [reply pkgs/servers/zigbee2mqtt/default.nix:14]: nit: s/binary/executable/

**Comment by zaninime** [comment]
Btw: I actually tweaked the `update.sh` script to run a `git checkout node.nix` after `node2nix` is invoked, as it was generating an invalid `node.nix`. The newly generated one was passing `utillinux` instead of `util-linux` to a downstream derivation.

**Comment by mweinelt** [comment]
> Btw: I actually tweaked the `update.sh` script locally to run a `git checkout node.nix` after `node2nix` is invoked, as it was generating an invalid `node.nix`. The newly generated one was passing `utillinux` instead of `util-linux` to a downstream derivation.

Yeah, that was the reason why I didn't update z2m to 1.16.2 already. Created an upstream issue over at https://github.com/svanderburg/node2nix/issues/213.

File: pkgs/servers/zigbee2mqtt/default.nix (hunk 3 above)
Comment (zaninime) [reply pkgs/servers/zigbee2mqtt/default.nix]: Good point. I [removed the `postInstall` completely](https://github.com/NixOS/nixpkgs/commit/b1776410b3c7ccaa6a3476e1bc5ea87cb4312882) and the NixOS test is running just fine.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105973` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zigbee2mqtt</li>
  </ul>
</details>

**Review by makefu — APPROVED** [review]
Deployed this version of zigbee2mqtt on my server at home and the network got kept alive!

**Pull request merged**